src/parser/view_transform.c

```diff
@@ -3162,6 +3207,68 @@ pt_copypush_terms (PARSER_CONTEXT * parser, PT_NODE * spec, PT_NODE * query, PT_
   return;
 }
 
+/*
+ * mq_copypush_sargable_terms_dblink() -
+ *   return:
+ *   parser(in):
+ *   statement(in):
+ *   spec(in):
+ *   new_query(in/out):
+ *   infop(in):
+ */
+static int
+mq_copypush_sargable_terms_dblink (PARSER_CONTEXT * parser, PT_NODE * statement, PT_NODE * spec, PT_NODE * new_query,
+				   FIND_ID_INFO * infop)
+{
+  PT_NODE *term, *new_term, *push_term_list;
+  int push_cnt, copy_cnt;
+  PT_NODE *save_next;
+
+  /* init */
+  push_term_list = NULL;
+  push_cnt = 0;
+  copy_cnt = -1;
+
+  for (term = statement->info.query.q.select.where; term; term = term->next)
+    {
+      if (pt_sargable_term (parser, term, infop) && PT_PUSHABLE_TERM (infop))
+	{
+	  /* copy term */
+	  new_term = parser_copy_tree (parser, term);
+
+	  /* for term, mark as copy-pushed term */
+	  if (term->node_type == PT_EXPR)
+	    {
+	      if (copy_cnt == -1)	/* very the first time */
+		{
+		  copy_cnt = pt_check_copypush_subquery (parser, new_query);
```

Comment:
There is nowhere to use copy_cnt. You need to confirm that it can be deleted.

Reply:
the next squence is to check if copy_cnt is zero. you can see the code " if (copy_cnt == 0)" at the next line.

Reply:
I will continue the discussion in the comments below.